### PR TITLE
Extract tests from Physical Infra Manager tests (#14)

### PR DESCRIPTION
**Pull request by caiocmpaes**
**What this PR does:**
- Extracts the PhysicaInfraManager tests from the single file to three other files:
  - `operations_spec.rb`
  - `manager_mixin_spec.rb`
  - `authenticatable_provider.rb`
  These files reflect the modules being tested.
- Moves the VCR Cassettes to match the specs folders.

**Reason:**
- Improve a separation of concerns and start a pattern where the `spec/models/` folder is a match to the `app/models/` folder.

### CODE REVIEW

**Comment by caiocmpaes** [comment]
@miq-bot add_label refactoring

**Comment by caiocmpaes** [comment]
@miq-bot assign @agrare 

@CharlleDaniel could you take a first look here, please?

**Comment by caiocmpaes** [comment]
@esdrasVP

**Comment by caiocmpaes** [comment]
@miq-bot remove_label unmergeable

File: app/models/manageiq/providers/lenovo/manager_mixin.rb [hunk 1]
```diff
@@ -9,11 +9,11 @@ module ManageIQ::Providers::Lenovo::ManagerMixin
   }.freeze
 
   def description
-    "Lenovo XClarity"
+    'Lenovo XClarity'
   end
 
   def connect(options = {})
-    # raise "no credentials defined" if missing_credentials?(options[:auth_type])
+    # raise 'no credentials defined' if missing_credentials?(options[:auth_type])
```
Comment (agrare) [inline app/models/manageiq/providers/lenovo/manager_mixin.rb]: @caiocmpaes did you rubocop -a the whole repo in this PR?

**Comment by agrare** [comment]
@caiocmpaes while I'm not against running rubocop -a on the whole repo:
1. I don't think it is that beneficial
2. It makes tracking history harder
3. It _definitely_ shouldn't be included in an unrelated PR, if anything lets do a single 'rubocop -a' PR and the team can review that by itself.  Trying to review 65 files of changes with a mixture of rubocop fixes and other stuff is insane

File: .rubocop_local.yml [hunk 2]
```diff
@@ -3,3 +3,14 @@ GlobalVars:
...
+# Ignores the file at lib folder that needs to have a naming with dashes.
+AllCops:
+  Exclude:
+  - lib/manageiq-providers-lenovo.rb
+
+# Forcing Single-quote string literals except in interpolations
+Style/StringLiterals:
+  Enabled: true
+Style/StringLiteralsInInterpolation:
+  Enabled: true
```
Comment (agrare) [inline .rubocop_local.yml]: Also we explicitly disable these for the MIQ source https://github.com/ManageIQ/guides/blob/master/.rubocop_base.yml#L235-L238 and shouldn't be enabled on a provider repo

**Comment by caiocmpaes** [comment]
@agrare I agree with you, Adam. Actually, that's exactly what I did. I opened 14 PRs with a step-to-step refactoring on the providers-lenovo.

The first PR is #173 and the PR title has a "# 1" at the end of the message. The PRs are ordered by dependence and the 5th PR (#181) is the one where I tried to fix all the rubocop issues.

This current PR is supposed to be the 14th of the sequence. When we merge the 1st commit (PR #173) the following PRs will get the first commit merged. Will this work as I expected?

File: .rubocop_local.yml (hunk 2 above)
Comment (caiocmpaes) [reply .rubocop_local.yml]: Ok. I will undo this change on PR #181

**Comment by caiocmpaes** [comment]
@miq-bot remove_label unmergeable

**Pull request merged**